### PR TITLE
feat: add further MovementAI skeleton

### PR DESCRIPTION
**Pull request by EmosewaMC**
tested that zone loading code loads and reads the correct values using logs.  other ldf data is unaffected as I walked around crux and dragons/apes can still spawn and be killed.

### CODE REVIEW

File: dGame/dComponents/TriggerComponent.cpp [hunk 1]
```diff
@@ -429,9 +441,50 @@ void TriggerComponent::HandleDestroySpawnerNetworkObjects(std::string args){
...
+	auto* movementAIComponent = targetEntity->GetComponent<MovementAIComponent>();
+	if (!movementAIComponent) return;
+	// movementAIComponent->SetupPath(argArray.at(0));
+	if (argArray.size() >= 2) {
+		auto index = GeneralUtils::TryParse<int32_t>(argArray.at(1));
+		if (!index) return;
+		// movementAIComponent->SetPathStartingWaypointIndex(index.value());
+	}
+	if (argArray.size() >= 3 && argArray.at(2) == "1") {
+		// movementAIComponent->ReversePath();
```
Comment (jadebenn) [inline dGame/dComponents/TriggerComponent.cpp:461]: Is this meant to be commented out?

File: dGame/dComponents/TriggerComponent.cpp [hunk 2]
```diff
@@ -429,9 +441,50 @@ void TriggerComponent::HandleDestroySpawnerNetworkObjects(std::string args){
...
+	}
+}
+
+void TriggerComponent::HandleTurnAroundOnPath(Entity* targetEntity) {
+	auto* movementAIComponent = targetEntity->GetComponent<MovementAIComponent>();
+	// if (movementAIComponent) movementAIComponent->ReversePath();
```
Comment (jadebenn) [inline dGame/dComponents/TriggerComponent.cpp:467]: Ditto

File: dGame/dComponents/TriggerComponent.cpp [hunk 3]
```diff
@@ -429,9 +441,50 @@ void TriggerComponent::HandleDestroySpawnerNetworkObjects(std::string args){
...
+}
+
+void TriggerComponent::HandleGoForwardOnPath(Entity* targetEntity) {
+	auto* movementAIComponent = targetEntity->GetComponent<MovementAIComponent>();
+	if (!movementAIComponent) return;
+	// if (movementAIComponent->GetIsInReverse()) movementAIComponent->ReversePath();
```
Comment (jadebenn) [inline dGame/dComponents/TriggerComponent.cpp:473]: If it is intentional (which it seems to be): Why?

Comment (EmosewaMC) [reply dGame/dComponents/TriggerComponent.cpp:473]: It's a skeleton from a different pr with lots of changes.  This code works in isolation and does not require the other half to function.  The commented out code is a note as to what will be called.

Comment (EmosewaMC) [reply dGame/dComponents/TriggerComponent.cpp:473]: None of these methods or functionality exist and as such this code just needs to exist and be functional in small chunks

File: dGame/dComponents/TriggerComponent.cpp [hunk 4]
```diff
@@ -164,8 +177,7 @@ std::vector<Entity*> TriggerComponent::GatherTargets(LUTriggers::Command* comman
 	std::vector<Entity*> entities = {};
 
 	if (command->target == "self") entities.push_back(m_Parent);
-	else if (command->target == "zone") { /*TODO*/ }
-	else if (command->target == "target" && optionalTarget) entities.push_back(optionalTarget);
+	else if (command->target == "zone") { /*TODO*/ } else if (command->target == "target" && optionalTarget) entities.push_back(optionalTarget);
```
Comment (aronwk-aaron) [inline dGame/dComponents/TriggerComponent.cpp:180]: two else's on one line

File: dCommon/dEnums/eWaypointCommandType.h [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	CHANGE_WAYPOINT,
+	DELETE_SELF,
+	KILL_SELF,
+	SPAWN_OBJECT,
+	PLAY_SOUND,
+};
+
+class WaypointCommandType {
+public:
+	static eWaypointCommandType StringToWaypointCommandType(std::string commandString) {
```
Comment (aronwk-aaron) [inline dCommon/dEnums/eWaypointCommandType.h:30]: can magic enum not be used to ease this?

Comment (EmosewaMC) [reply dCommon/dEnums/eWaypointCommandType.h:30]: Nope, recall this is data from lutriggers. And Aldo it's from string to enum, not enum to string.

File: dGame/dComponents/TriggerComponent.cpp (hunk 4 above)
Comment (EmosewaMC) [reply dGame/dComponents/TriggerComponent.cpp:180]: You wrote this :P

Comment (aronwk-aaron) [reply dGame/dComponents/TriggerComponent.cpp:180]: and I can't be wrong?

**Review by aronwk-aaron — DISMISSED** [review]

**Review by aronwk-aaron — APPROVED** [review]

**Review by jadebenn — APPROVED** [review]

**Pull request merged**